src/main/java/com/telefonica/iot/cygnus/sinks/OrionMongoSink.java

```diff
@@ -115,7 +117,8 @@ void persist(Map<String, String> eventHeaders, NotifyContextRequest notification
                 
                 // create the collection at this stage, if the data model is collection-per-attribute
                 if (dataModel == DataModel.COLLECTIONPERATTRIBUTE) {
-                    collectionName = buildCollectionName(fiwareServicePaths[i], destinations[i], attrName);
+                    collectionName = buildCollectionName(dbName, fiwareServicePaths[i], destinations[i], attrName,
+                            false, entityId, entityType, fiwareService);
                     backend.createCollection(dbName, collectionName);
```

Comment:
In case of hashing and in case the available bytes for the hash generation are not enough, the collectionName may be null. Does the createCollection() method deal this case properly? Thanks!